### ISSUE TITLE
Entry Text changing to black on slider change when is light mode iOS

### ISSUE THREAD

**Issue by ISSPRO-Eng** [dotnet/maui#21731]
### Description

When I am in light mode display in iOS, the entry that is connected to my sliders is changed from the set text color white to black. In dark mode, the text stays white. I have already changed the styles.xaml to only use white text color and even set the text color in the pages xaml. This behavior is not seen in Android. I do not even have any references to TextColor = "Black" anywhere in my project for an Entry. I deleted the Entry Property from the styles.xaml file in the resources to ensure that was not causing the issue.

### Steps to Reproduce

1. Create an entry that is also tied to a slider.
2. Slide the slider from the start position to a new position.
3. Observe the text color change from white to black
4. Delete the black text and enter a number
5. The number is entered as white

### Link to public reproduction project repository

This is an example snippet of where the issue lies.
`<Slider Maximum="255" Minimum="0"
                                Margin="10,0,10,0"
                                ThumbColor="{x:StaticResource Error}"
                                MinimumTrackColor="{x:StaticResource Error}"
                                MaximumTrackColor="{StaticResource Gray}"
                                ValueChanged="OnSliderValueChanged"
                                Value="{Binding RedVal}"/>
'
                        <StackLayout Orientation="Horizontal" HorizontalOptions="CenterAndExpand">   
                            <Label Text="{Binding GreenSliderLbl}"
                                   Style="{StaticResource LabelStyle}"
                                   VerticalOptions="Center"
                                   Grid.Column="0"
                                   />
                            <Entry Text="{Binding GreenVal, Converter={StaticResource NullableIntConverter}}"
                                   Style="{StaticResource EntryStyle}"
                                   Grid.Column="1"
                                   />
                            <Label Text=")"
                                   Style="{StaticResource LabelStyle}"
                                   VerticalOptions="Center"
                                   Grid.Column="0"
                                   />
                        </StackLayout>`

If I change the Entry to a Label, the text color remains white all the time. When I make it an entry, it changes to black if I slide the slider.

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI, Yes, this used to work in Xamarin.Forms

### Last version that worked well

7.0.100

### Affected platforms

iOS

### Affected platform versions

iOS 17.2

### Did you find any workaround?

I force the iOS to use the Dark Theme and the text color is white as expected.

**Comment by PureWeen** [comment]
Can you test with the latest nightly build? 
https://github.com/dotnet/maui/wiki/Nightly-Builds

**Comment by SkytteSam** [comment]
Hi @PureWeen
I had a similar problem where the Text where Black or White, depending on light/dark mode on iOS. When the text in the entry was set over the Binding from ViewModel. The set text color was ignored, but when I changed the text in the entry it changed back to the selected text. With the stable release of Microsoft.Maui.Controls 8.0.20 has this behaviour. 

Tested with the nightly build 8.0.40-nightly.10467, it works as expected. The entry's text colour is also used when the text is changed in the ViewModel.
